meilisearch-http/src/routes/api_key.rs

```diff
@@ -46,20 +47,21 @@ pub async fn create_api_key(
 
 pub async fn list_api_keys(
     auth_controller: GuardedData<ActionPolicy<{ actions::KEYS_GET }>, AuthController>,
-    _req: HttpRequest,
+    paginate: web::Query<Pagination>,
 ) -> Result<HttpResponse, ResponseError> {
-    let res = tokio::task::spawn_blocking(move || -> Result<_, AuthControllerError> {
+    let page_view = tokio::task::spawn_blocking(move || -> Result<_, AuthControllerError> {
         let keys = auth_controller.list_keys()?;
-        let res: Vec<_> = keys
-            .into_iter()
-            .map(|k| KeyView::from_key(k, &auth_controller))
-            .collect();
-        Ok(res)
+        let page_view = paginate.auto_paginate_sized(
+            keys.into_iter()
+                .map(|k| KeyView::from_key(k, &auth_controller)),
+        );
```

Comment:
Pretty cool that rust is able to keep the impl of ExactSizeIterator after a map :astonished:

Reply:
Yup! [`ExactSizeIterator` is impl on `Map<I>` if `I: ExactSizeIterator`](https://doc.rust-lang.org/std/iter/struct.Map.html#impl-ExactSizeIterator).